packages/cli/src/config/settings.ts

```diff
@@ -284,15 +294,44 @@ export function setUpCloudShellEnvironment(envFilePath: string | null): void {
   }
 }
 
-export function loadEnvironment(): void {
+export function loadEnvironment(settings?: Settings): void {
   const envFilePath = findEnvFile(process.cwd());
 
-  if (process.env.CLOUD_SHELL === 'true') {
-    setUpCloudShellEnvironment(envFilePath);
+  // Cloud Shell environment variable handling
+  if (process.env.CLOUDSHELL_ENVIRONMENT) {
```

Comment:
can you continue to use the setupCloudShellEnvironment? this change seems unrelated to the fix in this pr. sorry if I am missing something.
if changes need to be made to setUpCloudShellEnvironment please make them in that method.